peer/peerstorage.go

```diff
@@ -0,0 +1,128 @@
+package peer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+var (
+	// peerStorage is the key that maps the peer's storage to its
+	// public key in the Db.
+	peerStorage = []byte("peer-storage")
+
+	// ErrUninitializedDB is called when the DB is not initialized yet
+	// but a caller calls the `RetrievePeerData`.
+	ErrUninitializedDB = errors.New("uninitialized Kvdb peer data " +
+		"store")
+)
+
+// KvdbPeerStorage is the kvdb implementation of the DataStoreForPeer interface.
+type KvdbPeerStorage struct {
+	// DB is th kvdb backend for the storage layer.
+	DB kvdb.Backend
+}
+
+// KvdbPeerStorageConfig is the configuration detail for the kvdb storage layer.
+type KvdbPeerStorageConfig struct {
+	// DB is th kvdb backend for the storage layer.
+	DB kvdb.Backend
+}
+
+// NewKvdbPeerStorage returns an instance of the kvdb storage layer.
+func NewKvdbPeerStorage(c *KvdbPeerStorageConfig) (*KvdbPeerStorage, error) {
+	return &KvdbPeerStorage{
+		DB: c.DB,
+	}, nil
+}
```

Comment:
Yeah this is unnecessary. Just take in the DB backend as an argument and call it a day. No need to go through the ceremony of making a config wrapper.